healme.lic

```diff
@@ -69,32 +73,44 @@ class HealMe
 
   def heal_wound(spells, part, internal)
     if @using_waggle
-      spells['Heal Wounds']['cast'] = "cast #{part}"
-      spells['Heal Wounds']['cast'] = "cast #{part} internal" if internal
-      check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']}"
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']} internal" if internal
+      data = check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
+      if data
+        maxmana = part.sum{ |x| if x['type'] =~ /Fresh/i;x['level'];else;0;end;} * 2
+        if data['cambrinth'].flatten.sum >= maxmana
+          calculate_mana(1, maxmana, UserVars.discerns[spells['Heal Wounds']['abbrev']], spells['Heal Wounds']['cyclic'], @settings)
+        end
+      end
       loop do
         Flags.reset('hm-partial-heal')
         DRCA.cast_spell(spells['Heal Wounds'], @settings, true)
         echo "partial: #{Flags['hm-partial-heal']}" if UserVars.healme_debug
         break if !Flags['hm-partial-heal']
       end
     else
-      pause 0.5 while cast_spell('HW', @settings.empath_healing['HW'], part, internal)
+      pause 0.5 while cast_spell('HW', @settings.empath_healing['HW'], part.first['part'], internal)
     end
   end
 
   def heal_scar(spells, part)
     if @using_waggle
-      spells['Heal Scars']['cast'] = "cast #{part}"
-      check_discern(spells['Heal Scars'], @settings) if spells['Heal Scars']['use_auto_mana']
+      spells['Heal Scars']['cast'] = "cast #{part.first['part']}"
```

Comment:
❓ If I'm reading this right, you're only casting on the first part, but calculating mana based on all parts? That's not what I would expect, could you help me understand the mana calculation here?

Reply:
When you heal the wound first, it causes a scar. So wound external is 1, scar external is 1. Heal the external, scar external goes to 2.  So calcing across entire value, lets you heal for correct amount. Could multiply by 4 and only grab scars, but was less code to just grab all parts